### ISSUE TITLE
Disable building of executables in dependencies

### ISSUE THREAD

**Issue by robdockins** [commercialhaskell/stack#2050]
I have a situation where I depend on the library component of some packages, but don't want to build their executable components (because I don't need them, and they fail to compile for reasons I'm not interested in fixing).

I haven't been able to find any configuration knobs that will let me do this.

**Comment by mgsloan** [comment]
Yeah, this'd be nice.  Note that unfortunately there are already Cabal issues that make it difficult to omit building executables - https://github.com/commercialhaskell/stack/issues/1406 .  I think it is likely possible to work around it, but I haven't had time to investigate deeply anytime recently.
